syne_tune/blackbox_repository/conversion_scripts/scripts/yahpo_import.py

```diff
@@ -79,15 +112,74 @@ def __init__(self, benchmark):
         )
         self.num_seeds = 1
+        self._is_iaml = _check_whether_iaml(benchmark)
+        self._is_rbv2 = _check_whether_rbv2(benchmark)
+        self._initialize_for_scenario()
+        # Has to be called after `_initialize_for_scenario`, in order to
+        # transform fidelity space for some of the YAHPO scenarios
+        self._adjust_fidelity_space(fidelities)
+
+    def _initialize_for_scenario(self):
+        if self._is_iaml or self._is_rbv2:
+            # For `iaml_`, the fidelity `trainsize` has been evaluated at values
+            # [0.05, 0.1, 0.2, 0.4, 0.6, 0.8, 1]. We multiply these values by 20
+            # in order to obtain integers: [1, 2, 4, 8, 12, 16, 20]
+            # For `rbv2_`, the fidelity `trainsize` has been evaluated at values
+            # [0.05, 0.1, 0.15, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1]. We
+            # multiply these values by 20 in order to obtain integers:
+            # [1, 2, 3, 4, 6, 8, 10, 12, 14, 16, 18, 20]
+            if self._is_iaml:
+                assert len(self.fidelity_space) == 1
+            domain = self.fidelity_space.get("trainsize")
+            assert domain is not None
+            assert isinstance(domain, cs.Float)
+            assert domain.upper == 1 and domain.lower <= 0.05
+            self.fidelity_space["trainsize"] = cs.randint(1, 20)
+            if self._is_rbv2:
+                # For `rbv2_`, a second fidelity is `repl`, but it is constant
+                # 10, so can be removed
+                assert len(self.fidelity_space) == 2
+                assert "repl" in self.fidelity_space
+                del self.fidelity_space["repl"]
+
+    def _adjust_fidelity_space(self, fidelities: Optional[List[int]]):
+        assert len(self.fidelity_space) == 1, "Only one fidelity is supported"
+        self._fidelity_name, domain = next(iter(self.fidelity_space.items()))
+        assert (
+            domain.value_type == int
+        ), f"value_type of fidelity attribute must be int, but is {domain.value_type}"
+        if fidelities is None:
+            self._fidelity_values = np.arange(domain.lower, domain.upper + 1)
+        else:
+            assert is_increasing(fidelities) and is_positive_integer(
+                fidelities
+            ), f"fidelities = {fidelities} must be strictly increasing positive integers"
+            assert (
+                domain.lower <= fidelities[0] and fidelities[-1] <= domain.upper
+            ), f"fidelities = {fidelities} must lie in [{domain.lower}, {domain.upper}]"
+            self._fidelity_values = np.array(fidelities)
+            self.fidelity_space[self._fidelity_name] = cs.ordinal(fidelities.copy())
 
     def _objective_function(
         self,
-        configuration: Dict,
-        fidelity: Optional[Dict] = None,
+        configuration: dict,
+        fidelity: Optional[dict] = None,
         seed: Optional[int] = None,
-    ) -> Dict:
+    ) -> dict:
```

Comment:
You'd have to explain. `dict` is a built-in type, no?

Reply:
AFAIK python version before 3.9 do not support parametrized generics (e.g. `dict[str, str]`) meaning a typing module needs to be used (e.g. `typing.Dict[str, str]`) for type hints with parametrized data types. In this case we are not using parameters (`configuration: dict` rather than `configuration: dict[str, str]`) so it makes no difference. However, I think it would be best practice to either use the future annotations import (`from __future__ import annotations`) or revert to using typing so that modifying those annotations with data dypes wouldnt break python 3.7+3.8.

Reply:
Sure, but we are using `dict` and not `dict[str, ...]` here.

I am just against writing `Dict` on its own, just like you'd not write `List` or `Tuple` on its own.

Reply:
I understand that the convention here is to use `typing` for parametrized generics and build-in types for non-parametrized generics. I havent seen that before but its fine by me.